docs/sdk-and-tools/sdk-js/sdk-js-cookbook-v13.md

```diff
@@ -0,0 +1,353 @@
+console.log("TX hash:", txHash); 
+```
+
+### Wait for transaction completion
+
+```
+import { TransactionWatcher } from "@multiversx/sdk-core";
+
+const watcherUsingApi = new TransactionWatcher(apiNetworkProvider);
+const transactionOnNetworkUsingApi = await watcherUsingApi.awaitCompleted(txHash);
+```
+
+If, instead, you use a `ProxyNetworkProvider` to instantiate the `TransactionWatcher`, you'll need to patch the `getTransaction` method,
+so that it instructs the network provider to fetch the so-called _processing status_, as well (required by the watcher to detect transaction completion).
+
+```
+const watcherUsingProxy = new TransactionWatcher({
+    getTransaction: async (hash) => { return await proxyNetworkProvider.getTransaction(hash, true) }
+});
+
+const transactionOnNetworkUsingProxy = await watcherUsingProxy.awaitCompleted(txHash);
+```
+
+In order to wait for multiple transactions:
+
+```
+await Promise.all([
+    watcherUsingApi.awaitCompleted(tx1), 
+    watcherUsingApi.awaitCompleted(tx2), 
+    watcherUsingApi.awaitCompleted(tx3)
+]);
+```
+
+For a different awaiting strategy, also see [extending sdk-js](https://docs.multiversx.com/sdk-and-tools/sdk-js/extending-sdk-js).
+## Token transfers
+
+Generally speaking, in order to create transactions that transfer native tokens or ESDT tokens, one should use the `TransferTransactionsFactory` class.
+
+::note
+In `sdk-core v13`, the `TransferTransactionsFactory` class was extended with new methods, 
+to be aligned with the [SDKs specs](https://github.com/multiversx/mx-sdk-specs/blob/main/core/transactions-factories/transfer_transactions_factory.md).
+The old, legacy methods are still available (see below), thus existing client code isn't affected.
+:::
+
+:::note
+In `sdk-core v13`, the `TokenTransfer` class has changed, in a non-breaking manner. 
+Though, from now on, it should only be used for prepairing ESDT token transfers, not native EGLD transfers.
+
+A `TokenTransfer` object can be instantiated using the legacy approaches, e.g. `fungibleFromAmount`, `nonFungible` (which are still available), 
+or with the new approach (which, among others, makes abstraction of the number of decimals a token has).
+:::
+
+:::tip
+For formatting or parsing token amounts, see [formatting and parsing amounts](#formatting-and-parsing-amounts).
+:::
+
+First, let's create a `TransferTransactionsFactory`:
+
+```
+import { GasEstimator, Token, TokenComputer, TokenTransfer, TransactionsFactoryConfig, TransferTransactionsFactory } from "@multiversx/sdk-core";
+
+// The new approach of creating a "TransferTransactionsFactory":
+const factoryConfig = new TransactionsFactoryConfig({ chainID: "D" });
+const factory = new TransferTransactionsFactory({ config: factoryConfig, tokenComputer: new TokenComputer() });
+
+// The legacy approach of creating a "TransferTransactionsFactory":
+const legacyFactory = new TransferTransactionsFactory(new GasEstimator());
+```
```

Comment:
if you agree, same comment as above, drop the legacy.

Reply:
Fixed.